### PR TITLE
support and default to compact schedule format

### PR DESCRIPTION
**Pull request by guanyilun**
The currently supported toast schedule format is produced with the verbose mode which contains more information than needed in operation. In this PR, I added support for the compact schedule format and use it as the default schedule format. This means that old schedule files will no longer be compatible without adding a `verbose=True` option in the `parse_sequence_from_toast` function call.

### CODE REVIEW

**Review by BrianJKoopman — COMMENTED** [review]
This looks alright to me, however when I go to test it with one of the compact schedules that were generated I get the following:
```
DEBUG:scheduler_server.app:Processing request: {'t0': '2023-08-14 12:00', 't1': '2023-08-14 14:00', 'policy': '{"policy": "basic", "config": {}}'}
ERROR:scheduler_server.app:Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/scheduler_server/app.py", line 98, in schedule
    commands = POLICY_HANDLERS[policy_name](t0, t1, policy_config, app.config)
  File "/usr/local/lib/python3.10/dist-packages/scheduler_server/handler.py", line 27, in basic_policy
    seq = policy.init_seqs(t0, t1)
  File "/usr/local/lib/python3.10/dist-packages/schedlib/policies.py", line 50, in init_seqs
    master = utils.parse_sequence_from_toast(self.master_schedule)
  File "/usr/local/lib/python3.10/dist-packages/schedlib/utils.py", line 105, in parse_sequence_from_toast
    throw=np.abs(row['az_max'] - row['az_min']),
TypeError: unsupported operand type(s) for -: 'str' and 'str'
```

I'm testing with the first schedule in the table on [this page from 2023-08-04](https://simonsobs.atlassian.net/wiki/spaces/PRO/pages/139034926/SAT+Schedules#2023-08-04).

Here's the query on the currently release scheduler-server:
```
$ curl -L -X POST -H "Content-Type: application/json" -d '{"t0": "2023-08-14 12:00", "t1": "2023-08-14 14:00", "policy": "{\"policy\": \"basic\", \"config\": {}}"}' http://127.0.0.1:8010/api/v1/schedule | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 28   368    0     0  100   105      0  16566 --:--:-- --:--:-- --:--:-- 17500
100  1672  100  1567  100   105   1836    123 --:--:-- --:--:-- --:--:--     0
{
  "commands": "from sorunlib import *\nfrom nextline import disable_trace\n\nwith disable_trace():\n    initialize(test_mode=True)\n\nsmurf.uxm_setup()\nsmurf.iv_curve()\n\n# DEC-050..-030_RA+046.452..+058.065\nacu.move_to(218.02, 50.00)\nsmurf.bias_dets()\nwait_until('2023-08-14T12:18:00+00:00')\nsmurf.bias_step()\nseq.scan(description='DEC-050..-030_RA+046.452..+058.065', stop_time='2023-08-14T12:38:56.998519+00:00', width=32.27)\nsmurf.bias_step()\n\n# uranus\nacu.move_to(317.55, 36.65)\nsmurf.bias_dets()\nwait_until('2023-08-14T12:38:56.998519+00:00')\nsmurf.bias_step()\nseq.scan(description='uranus', stop_time='2023-08-14T12:45:22.500119+00:00', width=1.25)\nsmurf.bias_step()\n\n# DEC-050..-030_RA+046.452..+058.065\nacu.move_to(218.02, 50.00)\nsmurf.bias_dets()\nwait_until('2023-08-14T12:45:22.500119+00:00')\nsmurf.bias_step()\nseq.scan(description='DEC-050..-030_RA+046.452..+058.065', stop_time='2023-08-14T13:03:30+00:00', width=32.27)\nsmurf.bias_step()\n\n# DEC-050..-030_RA+046.452..+058.065\nacu.move_to(218.02, 50.00)\nsmurf.bias_dets()\nwait_until('2023-08-14T13:03:30+00:00')\nsmurf.bias_step()\nseq.scan(description='DEC-050..-030_RA+046.452..+058.065', stop_time='2023-08-14T13:31:00+00:00', width=32.25)\nsmurf.bias_step()\n\n# DEC-050..-030_RA+069.677..+081.290\nacu.move_to(217.94, 50.00)\nsmurf.bias_dets()\nwait_until('2023-08-14T13:51:00+00:00')\nsmurf.bias_step()\nseq.scan(description='DEC-050..-030_RA+069.677..+081.290', stop_time='2023-08-14T14:00:00+00:00', width=32.28)\nsmurf.bias_step()\n",
  "message": "Success",
  "status": "ok"
}
```

And the same query on a test build:
```
$ curl -L -X POST -H "Content-Type: application/json" -d '{"t0": "2023-08-14 12:00", "t1": "2023-08-14 14:00", "policy": "{\"policy\": \"basic\", \"config\": {}}"}' http://127.0.0.1:8011/api/v1/schedule | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 28   368    0     0  100   105      0   1781 --:--:-- --:--:-- --:--:--  1810
100   194  100    89  100   105    684    807 --:--:-- --:--:-- --:--:--  1491
{
  "message": "Error: unsupported operand type(s) for -: 'str' and 'str'",
  "status": "error"
}
```

EDIT: It's been a while since I've worked with pandas, but it seems `dtypes` can be passed based on [this blog post](https://saturncloud.io/blog/pandas-readfwf-specifying-dtype/), however I can't find that in their official documentation.

**Comment by guanyilun** [comment]
ah I see, it's the fix-width format being broken by the patch name, such as
```
  2023-08-06 06:00:00  2023-08-06 07:05:00     0.00 SPECIAL -- Moon avoidance (at end)    110.00   130.00    50.00     3    5
  2023-08-06 07:05:00  2023-08-06 08:10:00     0.00 SPECIAL -- Moon avoidance (at beginning) 110.00   130.00    50.00     3    6
```
Given the number of spaces in each field, it's hard to parse without knowing the width ahead of time. Maybe we should better stick to a fixed width in naming patches.

**Comment by BrianJKoopman** [comment]
New schedule files with the '|' delimiter are [posted here](https://simonsobs.atlassian.net/wiki/spaces/PRO/pages/139034926/SAT+Schedules#2023-08-18-%3A-Hit-maps%2C-delimiters-and-%E2%80%9Ctuning%E2%80%9D-timing-updates). Let me know if there's any trouble parsing these.

**Review by BrianJKoopman — APPROVED** [review]
This looks good, thanks!

One small thing, I'd encourage you to add at least one of the "SPECIAL -- Moon avoidance (at end)" scans to the test file, since that's what broke parsing prior to switching to the `|` delimited format.
